eng/scripts/New-ChangelogEntry.ps1

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env pwsh
+<#
+.SYNOPSIS
+    Creates a new changelog entry YAML file.
+
+.DESCRIPTION
+    This script helps create properly formatted changelog entry files.
+    Each entry is stored as a separate YAML file to avoid merge conflicts.
+
+.PARAMETER Description
+    Description of the change (minimum 10 characters).
+
+.PARAMETER Section
+    The changelog section. Valid values: "Features Added", "Breaking Changes", "Bugs Fixed", "Other Changes".
+
+.PARAMETER Subsection
+    Optional subsection for grouping related changes (e.g., "Dependency Updates", "Telemetry").
+
+.PARAMETER PR
+    Pull request number (integer).
+
+.PARAMETER ServerName
+    Name of the server to create changelog entry for (e.g., "Azure.Mcp.Server", "Fabric.Mcp.Server").
+    Defaults to "Azure.Mcp.Server".
+
+.PARAMETER ChangelogEntriesPath
+    Path to the changelog-entries directory. If not specified, uses servers/{ServerName}/changelog-entries.
+
+.EXAMPLE
+    ./eng/scripts/New-ChangelogEntry.ps1
+
+    Runs in interactive mode for Azure.Mcp.Server, prompting for all required fields.
+
+.EXAMPLE
+    ./eng/scripts/New-ChangelogEntry.ps1 -ServerName "Fabric.Mcp.Server"
+
+    Runs in interactive mode for Fabric.Mcp.Server.
+
+.EXAMPLE
+    ./eng/scripts/New-ChangelogEntry.ps1 -Description "Added new feature" -Section "Features Added" -PR 1234
+
+    Creates a changelog entry for Azure.Mcp.Server with the specified parameters.
+
+.EXAMPLE
+    ./eng/scripts/New-ChangelogEntry.ps1 -ServerName "Fabric.Mcp.Server" -Description "Updated Azure.Core to 1.2.3" -Section "Other Changes" -Subsection "Dependency Updates" -PR 1234
+
+    Creates a changelog entry for Fabric.Mcp.Server with a subsection.
+
+.EXAMPLE
+    $description = @"
+Added new AI Foundry tools:
+- foundry_agents_create: Create a new AI Foundry agent
+- foundry_threads_create: Create a new AI Foundry Agent Thread
+- foundry_threads_list: List all AI Foundry Agent Threads
+"@
+    ./eng/scripts/New-ChangelogEntry.ps1 -Description $description -Section "Features Added" -PR 945
+
+    Creates a changelog entry with a multi-line description containing a list.
+#>
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $false)]
+    [string]$ServerName = "Azure.Mcp.Server",
+
+    [Parameter(Mandatory = $false)]
+    [string]$Description,
+
+    [Parameter(Mandatory = $false)]
+    [string]$Section,
+
+    [Parameter(Mandatory = $false)]
+    [string]$Subsection,
+
+    [Parameter(Mandatory = $false)]
+    [int]$PR,
+
+    [Parameter(Mandatory = $false)]
+    [string]$ChangelogEntriesPath
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = "Stop"
+
+# Set default path based on ServerName
+if (-not $ChangelogEntriesPath) {
+    $ChangelogEntriesPath = "servers/$ServerName/changelog-entries"
```

Comment:
Rather than making this server specific, can we have it target the changelog file?  i.e.  This script could manage any `CHANGELOG.md` file with a `changelog-entries` directory beside it.